src/wp-includes/option.php

```diff
@@ -827,7 +822,17 @@ function update_option( $option, $value, $autoload = null ) {
 	);
 
 	if ( null !== $autoload ) {
-		$update_args['autoload'] = ( 'no' === $autoload || false === $autoload ) ? 'no' : 'yes';
+		$autoload                = determine_option_autoload_value( $option, $serialized_value, $autoload );
+		$update_args['autoload'] = $autoload;
```

Comment:
From looking at those two specific lines, I would agree with you. However, there's also the concern about the `$autoload` variable not carrying the correct value - which doesn't cause any problem in the way the code is now, but it's IMO error-prone. Someone may change the code further down in the function to rely on `$autoload` somewhere, assuming that's fine to use (because why wouldn't it). For that reason I would personally lean towards updating the variable value here too. I don't feel strongly about it, but think it's a bit safer.

Reply:
Uh, yea, that would be a really poorly written code though :) 

In any case, not that important, and having a single use var might be helping readability a little bit.